app/models/tenant.rb

```diff
@@ -39,8 +39,13 @@ def to_s
   attr_accessible :incorporated_on
 
   def fiscal_period(year)
-    month = fiscal_year_ends_on.try(:month) || 12
-    day = fiscal_year_ends_on.try(:day) || 31
+    if fiscal_year_ends_on && fiscal_year_ends_on.year == year
```

Comment:
There are companies which do not have their business year matching calendar years. So their business year 2016 might start on 2015-04-01 and end on 2016-03-31.

We support this by allow specifying the end of the business/fiscal year. It is used in the year filters as book keeper normally focus on business years, not calendar years. The main exceptions are VAT and salary reports.